package/network/services/hostapd/files/hostapd.sh

```diff
@@ -627,7 +627,7 @@ wpa_supplicant_add_network() {
 		scan_ssid=""
 	}
 
-	[[ "$_w_mode" = "adhoc" -o "$_w_mode" = "mesh" ]] && append network_data "$_w_modestr" "$N$T"
+	[[ "$_w_mode" = "adhoc" || "$_w_mode" = "mesh" ]] && append network_data "$_w_modestr" "$N$T"
```

Comment:
Would you please move this change into an own commit!

This is the only change related to bash compatibility (LEDE uses busybox ash by default). All other changes are fixing common syntax/typos/logic errors.

I would suggest to use the following commit titles or similar:

- treewide: fix shellscript syntax errors/typos
- hostapd: fix bash compatibility

Reply:
I decided to remove useless [ ] instead of replace -o with ||